hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/ReadResultSetImpl.java

```diff
@@ -88,46 +108,65 @@ public int readCount() {
 
     @Override
     public void setHazelcastInstance(HazelcastInstance hz) {
-        this.hz = hz;
+        setSerializationService(((SerializationServiceSupport) hz).getSerializationService());
+    }
+
+    public void setSerializationService(SerializationService serializationService) {
+        this.serializationService = serializationService;
     }
 
     @Override
     public E get(int index) {
         checkNotNegative(index, "index should not be negative");
         checkTrue(index < size, "index should not be equal or larger than size");
 
-        SerializationService serializationService = getSerializationService();
-
-        Data item = items[index];
+        final Data item = items[index];
         return serializationService.toObject(item);
     }
 
-    private SerializationService getSerializationService() {
-        HazelcastInstanceImpl impl = (HazelcastInstanceImpl) hz;
-        return impl.getSerializationService();
-    }
-
-    public void addItem(Object item) {
+    @Override
+    public long getSequence(int index) {
+        return seqs != null && seqs.length > index ? seqs[index] : -1;
```

Comment:
Then shouldn't `getSequence(index)` explicitly fail when version is 3.8, similar to other 3.9 features?